### ISSUE TITLE
Ångström Beagleboneblack Java Issue

### ISSUE THREAD

**Issue by midgard99** [usb4java/usb4java#36]
Hello,
I'm having some troubles in work with Java-linux-arm library, I created a project in Netbeans with usb4java version 1.2.0 using libusb4java-1.2.0-linux-arm.jar and windows-x86 64.jar to a comunication between Ångström Linux and Windows 7 64 but when I go to run the jar file in linux this appears:
The command used to run the Jar: java -Djava.io.tmpdir=/home/root/Desktop/temp -jar Java4USB.jar
That was to be certain that the folder have the rights to Write and Read.

If anyone can help I will be grateful,
Thanks in advice
midgard99

Error:

Exception in thread "main" java.lang.UnsatisfiedLinkError: /home/root/Desktop/temp/usb4java5571950167071539816.tmp/libusb4java.so: ld-linux-armhf.so.3: cannot open shared object file: No such file or directory
        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
        at java.lang.ClassLoader.loadLibrary1(ClassLoader.java:1965)
        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1890)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1851)
        at java.lang.Runtime.load0(Runtime.java:795)
        at java.lang.System.load(System.java:1062)
        at org.usb4java.Loader.load(Loader.java:358)
        at org.usb4java.LibUsb.<clinit>(LibUsb.java:640)
        at newpackage.main.main(main.java:35)

Code:

package newpackage;

import org.usb4java.Context;
import org.usb4java.Device;
import org.usb4java.DeviceDescriptor;
import org.usb4java.DeviceList;
import org.usb4java.LibUsb;
import org.usb4java.LibUsbException;

/**
- Simply lists all available USB devices.
- 
- @author 
  _/
  public class main
  {
  /_*
  - Main method.
  - 
  - @param args
  - ```
           Command-line arguments (Ignored)
    ```
    
    */
    public static void main(String[] args)
    {
    // Create the libusb context
    Context context = new Context();
    
    // Initialize the libusb context
    int result = LibUsb.init(context);
    if (result < 0)
    {
        throw new LibUsbException("Unable to initialize libusb", result);
    }
    
    // Read the USB device list
    DeviceList list = new DeviceList();
    result = LibUsb.getDeviceList(context, list);
    if (result < 0)
    {
        throw new LibUsbException("Unable to get device list", result);
    }
    
    try
    {
        // Iterate over all devices and list them
        for (Device device: list)
        {
            int address = LibUsb.getDeviceAddress(device);
            int busNumber = LibUsb.getBusNumber(device);
            DeviceDescriptor descriptor = new DeviceDescriptor();
            result = LibUsb.getDeviceDescriptor(device, descriptor);
            if (result < 0)
            {
                throw new LibUsbException(
                    "Unable to read device descriptor", result);
            }
            System.out.format(
                "Bus %03d, Device %03d: Vendor %04x, Product %04x%n",
                busNumber, address, descriptor.idVendor(),
                descriptor.idProduct());
        }
    }
    finally
    {
        // Ensure the allocated device list is freed
        LibUsb.freeDeviceList(list, true);
    }
    
    // Deinitialize the libusb context
    LibUsb.exit(context);
    }
    }

**Comment by allengregoryiv** [comment]
Was a solution ever found to this, I'm having the same `UnsatisfiedLinkError`

**Comment by kayahr** [comment]
usb4java 1.3.0 was released today which now uses libusb 1.0.22 which may have fixed this issue. Please try again with this new version and give feedback here if the exact same problem still occurs. If it still doesn't work but it is not the exact same result then please open a new issue.
